elastic/kibana
# [Fleet] Fix race condition in reassign modal

## Summary

Fixes https://github.com/elastic/kibana/issues/134898 and an issue where the selected policy in the dropdown was the not the policy that was actually sent to the API, resulting in attempting to reassign agents to a managed policy which is not allowed.

![image](https://user-images.githubusercontent.com/1813008/175533361-1bf01430-8815-481d-8600-a2d6c1398370.png)

### To test

1. Deploy this on cloud where there's a managed policy
2. Create a regular agent policy, then duplicate it
3. Enroll 2 agents in the first policy
4. Select both agents, open reassign bulk action
5. Observe that the integration badges are accurate
6. Click reassign, and verify reassignment succeeds

### To workaround

If you need to workaround this bug in a build that doesn't include this fix, you can select a new policy from the dropdown and then go back to the one you want and things should work fine.
